semgrep/semgrep/semgrep_main.py

```diff
@@ -255,6 +257,12 @@ def main(
             )
             optimizations = "none"
 
+    join_rules, rest_of_the_rules = partition(
+        lambda rule: rule.raw.get("mode") == JOIN_MODE,
```

Comment:
does rule.mode not give us what we want here?

Reply:
I think I thought that `rule.mode` may not be present if `mode:` wasn't explicitly present, but it looks like it gets automatically populated. I added a new e2e test that checks multiple rule modes in the same file to make sure it works as expected